design/src/docs/code_generation.md

```diff
@@ -0,0 +1,98 @@
+## Generating common service code
+
+How a service is constructed and how to plug in new business logic is described in [Pokémon Service][1].
+This document introduces the project and how code is being generated. It is written for developers who want to start contributing to `smithy-rs`.
+
+### Folder structure
+
+The project is divided in:
+
+- `/codegen`: it contains shared code for both client and server, but only generates a client
+- `/codegen-server`: server only. This project started with `codegen` to generate a client, but client and server share common code; that code lives in `codegen`, which `codegen-server` depends on
+- `/aws`: the AWS Rust SDK, it deals with AWS services specifically. The folder structure reflects the project's, with the `rust-runtime` and the `codegen`
+- `/rust-runtime`: the generated client and server crates may depend on crates in this folder. Crates here are not code generated
```

Comment:
Probably a good idea to talk about the special `inlinable` cargo project inside `/rust-runtime` since it's not actually a runtime crate.